File: generated-code/codecraft/swift/Sources/TransGenTest/Player.swift
public struct Player {
    let id: Int32
    let score: Int32
    let resource: Int32

    static func readFrom<S: InputStream>(_ stream: S) -> Player {
        var id: Int32
        id = stream.readInt32()
        var score: Int32
        score = stream.readInt32()
        var resource: Int32
        resource = stream.readInt32()
        return Player(id: id, score: score, resource: resource)
    }

    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(id)
        stream.writeInt32(score)
        stream.writeInt32(resource)
    }
}

File: generated-code/scala/codecraft/TcpReadWrite/src/main/scala/trans_gen_test/model/Player.scala
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Player (strategy, client)
 *
 * @param id Player's ID
 * @param score Current score
 * @param resource Current amount of resource
 */
case class Player(id: Int, score: Int, resource: Int) {
    /**
     * Write Player to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        StreamUtil.writeInt(stream, score)
        StreamUtil.writeInt(stream, resource)
    }

    /**
     * Get string representation of Player
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Player { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("score: ")
        stringBuilder.append(score)
        stringBuilder.append(", ")
        stringBuilder.append("resource: ")
        stringBuilder.append(resource)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Player {
    /**
     * Read Player from input stream
     */
    def readFrom(stream: java.io.InputStream): Player = Player(
        StreamUtil.readInt(stream),
        StreamUtil.readInt(stream),
        StreamUtil.readInt(stream)
    )
}